Deployment error: Missing `pandas` on `icenet-application` deployment - error

# Description of issue

When deploying [icenet-application](https://github.com/icenet-ai/icenet-application), Azure logs show a missing pandas dependency issue, even though it should be installed somewhere in the dependency tree of the available `requirements.txt`.

# Error

```bash
2024-10-02T15:31:47.903759390Z [notice] A new release of pip available: 22.3.1 -> 24.2
2024-10-02T15:31:47.903765890Z [notice] To update, run: python3.8 -m pip install --upgrade pip
2024-10-02T15:31:50.052461078Z [2024-10-02 15:31:50 +0000] [82] [INFO] Starting gunicorn 20.1.0
2024-10-02T15:31:50.062759716Z [2024-10-02 15:31:50 +0000] [82] [INFO] Listening at: http://0.0.0.0:8000 (82)
2024-10-02T15:31:50.062940119Z [2024-10-02 15:31:50 +0000] [82] [INFO] Using worker: sync
2024-10-02T15:31:50.092039308Z [2024-10-02 15:31:50 +0000] [84] [INFO] Booting worker with pid: 84
2024-10-02T15:31:50.145437824Z [2024-10-02 15:31:50 +0000] [84] [ERROR] Exception in worker process
2024-10-02T15:31:50.145468524Z Traceback (most recent call last):
2024-10-02T15:31:50.145473224Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/arbiter.py", line 589, in spawn_worker
2024-10-02T15:31:50.145476924Z     worker.init_process()
2024-10-02T15:31:50.145486024Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/workers/base.py", line 134, in init_process
2024-10-02T15:31:50.145489424Z     self.load_wsgi()
2024-10-02T15:31:50.145492424Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
2024-10-02T15:31:50.145495924Z     self.wsgi = self.app.wsgi()
2024-10-02T15:31:50.145499024Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/app/base.py", line 67, in wsgi
2024-10-02T15:31:50.145502224Z     self.callable = self.load()
2024-10-02T15:31:50.145505224Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
2024-10-02T15:31:50.145508424Z     return self.load_wsgiapp()
2024-10-02T15:31:50.145511424Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
2024-10-02T15:31:50.145514625Z     return util.import_app(self.app_uri)
2024-10-02T15:31:50.145517625Z   File "/opt/python/3.8.12/lib/python3.8/site-packages/gunicorn/util.py", line 359, in import_app
2024-10-02T15:31:50.145521125Z     mod = importlib.import_module(module)
2024-10-02T15:31:50.145524225Z   File "/opt/python/3.8.12/lib/python3.8/importlib/__init__.py", line 127, in import_module
2024-10-02T15:31:50.145527425Z     return _bootstrap._gcd_import(name[level:], package, level)
2024-10-02T15:31:50.145530525Z   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
2024-10-02T15:31:50.145534425Z   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
2024-10-02T15:31:50.145537525Z   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
2024-10-02T15:31:50.145540725Z   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
2024-10-02T15:31:50.145544325Z   File "<frozen importlib._bootstrap_external>", line 843, in exec_module
2024-10-02T15:31:50.145547825Z   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
2024-10-02T15:31:50.145551025Z   File "/home/site/wwwroot/icenet_app/app.py", line 7, in <module>
2024-10-02T15:31:50.145554425Z     import pandas as pd
2024-10-02T15:31:50.145557525Z ModuleNotFoundError: No module named 'pandas'
2024-10-02T15:31:50.145560725Z [2024-10-02 15:31:50 +0000] [84] [INFO] Worker exiting (pid: 84)
2024-10-02T15:31:50.284801590Z [2024-10-02 15:31:50 +0000] [82] [INFO] Shutting down: Master
2024-10-02T15:31:50.292295690Z [2024-10-02 15:31:50 +0000] [82] [INFO] Reason: Worker failed to boot.
```

# Fix
Thanks to @JimCircadian for figuring out that following needs enabling to allow the [Oryx](https://github.com/microsoft/Oryx) build system to build the dependencies from `requirements.txt` before deploying the app.

https://github.com/icenet-ai/icenet-etl/blob/d71284cfdf52811cc711a483c488c8c11ec2bc73/terraform/application/app.tf#L35

This only seems to happen when using `make deploy-zip` in [icenet-application](https://github.com/icenet-ai/icenet-application), and works without this change if using `make deploy-azure` which deploys from the `icenet-application` git repo directly instead of a local clone.

--- a/terraform/application/app.tf
+++ b/terraform/application/app.tf
@@ -32,7 +32,7 @@ resource "azurerm_linux_web_app" "this" {
     "ICENET_AUTH_LIST"        = "/data/auth_list.json"
     "ICENET_DATA_LOCATION"    = "/data"
 #    "ENABLE_ORYX_BUILD"              = "true"
-#    "SCM_DO_BUILD_DURING_DEPLOYMENT" = "true"
+    "SCM_DO_BUILD_DURING_DEPLOYMENT" = "true"
   }
 
   storage_account {

